src/core/device_manager.py

```
@@ -441,288 +572,353 @@ def _get_unauthorized_device_info(self, serial: str, metadata: Dict[str, str] =
+
         return props
-    
+
     def _check_frp_status(self, serial: str) -> str:
         """Check FRP status of device"""
         try:
             # Try multiple methods to check FRP status
-            
+
             # Method 1: Check persistent properties
             result = subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)